crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -903,22 +905,35 @@ async fn payment_response_update_tracker<F: Clone, T: types::Capturable>(
             },
             Err(_) => None,
         };
-        if let Some(ref payment_method) = payment_data.payment_method_info {
-            payments::tokenization::update_connector_mandate_details_in_payment_method(
+        if let Some(payment_method_id) = &m_payment_method_id {
+            let payment_method = state
+                .store
+                .find_payment_method(payment_method_id)
+                .await
+                .to_not_found_response(errors::ApiErrorResponse::PaymentMethodNotFound)?;
+
+            let connector_mandate_details =
+                payments::tokenization::update_connector_mandate_details_in_payment_method(
```

Comment:
This function should not have been an async function. This indicates that either some network call or database call might be happening in this function, (in this case a database call) but none of them are happening in this function.